### PR TITLE
Story/fire starts dropdown/1551

### PR DESCRIPTION
**Pull request by conbrad**
Story changes:
- Configurable fire starts in the weekly table view
- Adjusting the fire starts for a day retriggers HFI calculations
- Encodes lookup tables from fire start range and mean intensity -> prep level

Technical changes:
- Refactors redux state further, to improve the structure for all components
- Instead of flat arrays of dailies in the planning area map, the structure is now:
```
export interface PlanningAreaResult {
  highestDailyIntensityGroup: number
  meanPrepLevel: number | undefined
  dailyResults: DailyResult[]
}

export interface DailyResult {
  dateISO: string
  dailies: StationDaily[]
  fireStarts: FireStarts
  meanIntensityGroup: number | undefined
  prepLevel: number | undefined
}
```

Example of a calculate HFI result in the redux store would be:

```
{
  "Kamloops (K2)": {
    "highestDailyIntensityGroup": 2,
    "meanPrepLevel": 1,
    "dailyResults": [
      {
        "dateISO": "mon - some iso format",
        "dailies": [{ "foo": "bar" }],                      /* station dailies within the area for the above date */,
        "fireStarts": { "label": "0-1", "value": 1 },     /* fire starts for this planning area for the above date */,
        "meanIntensityGroup": 1,
        "prepLevel": 1
      },
      /* ... more dailies */ 
    ]
  }
}
````

# Test Links:
[Percentile Calculator](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1676.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Review by andrea-williams — CHANGES_REQUESTED** [review]
The math for (intensity group, daily starts) -> Prep Level doesn't appear to be correct.

Screen Shot 2022-01-31 at 8 58 39 AM

shows Prep Level 4 when starts are 6+ and Intensity Group is 4.5, but according to the HFI workbook, Prep Level should be 6.

Screen Shot 2022-01-31 at 8 59 45 AM

**Comment by conbrad** [comment]
> 6

Thank you! Will fix. I think I missed the changes needed for the PrepLevel Cell.

**Comment by conbrad** [comment]
> > 6
> 
> Thank you! Will fix. I think I missed the changes needed for the PrepLevel Cell.

Fixed

File: web/src/features/hfiCalculator/components/FireStartsDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  lowestFireStarts
+} from 'features/hfiCalculator/slices/hfiCalculatorSlice'
+import { isEqual, isNull } from 'lodash'
+import React from 'react'
+
+export interface FireStartsDropdownProps {
+  fireStarts: FireStarts | undefined
+  areaName: string
+  dayOffset: number
+  setFireStarts: (areaName: string, dayOffSet: number, newFireStarts: FireStarts) => void
```
Comment (andrea-williams) [inline web/src/features/hfiCalculator/components/FireStartsDropdown.tsx:15]:
```suggestion
  setFireStarts: (areaName: string, dayOffset: number, newFireStarts: FireStarts) => void
```

Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/FireStartsDropdown.tsx:15]: very picky & not really important - just suggesting it for consistency with how "Offset" is used elsewhere in the file

**Review by andrea-williams — APPROVED** [review]

**Review by tessmgood — APPROVED** [review]

File: web/src/features/hfiCalculator/components/FireStartsDropdown.tsx (hunk 1 above)
Comment (conbrad) [reply web/src/features/hfiCalculator/components/FireStartsDropdown.tsx]: Done

**Review by Sybrand — APPROVED** [review]

**Pull request merged**